quaistats/quaistats.go

```diff
@@ -588,6 +604,47 @@ func (s uncleStats) MarshalJSON() ([]byte, error) {
 	return []byte("[]"), nil
 }
 
+func (s *Service) computeTps(block *types.Block) (uint64, error) {
+	var parentTime, parentTps uint64
+
+	if parentCached, ok := s.tpsLookupCache.Get(block.ParentHash()); ok {
+		parentTime = parentCached.(blockCacheDto).Time
+		parentTps = parentCached.(blockCacheDto).Tps
+	} else {
+		log.Warn(fmt.Sprintf("block %d's parent not found in cache: %s", block.Number(), block.ParentHash().String()))
+		fullBackend, ok := s.backend.(fullNodeBackend)
+		if ok {
+			parent, err := fullBackend.BlockByNumber(context.Background(), rpc.BlockNumber(block.NumberU64()-1))
+			if err != nil {
+				return 0, errors.New(fmt.Sprintf("error getting parent block %s: %s", block.ParentHash().String(), err.Error()))
+			}
+			parentTime = parent.Time()
+			parentTps = UINT64_MAX_VALUE
+		} else {
+			return 0, errors.New("not running fullnode, cannot get parent block")
+		}
+	}
+
+	instantTps := uint64(len(block.Transactions()))
+	if dt := block.Time() - parentTime; dt != 0 {
+		instantTps /= dt // this is a hack to avoid dividing by 0
```

Comment:
Just to avoid confusion, might move the comment up. L630 isn't a hack. The `if ...; dt != 0` is the hack